pkg/snapstore/abs_snapstore.go

```diff
@@ -116,6 +115,23 @@ func getCredentials(prefixString string) (string, string, error) {
 		return credentials.StorageAccount, credentials.SecretKey, nil
 	}
 
+	// TODO: @renormalize Remove this extra handling in v0.31.0
+	// Check if a JSON file is present in the directory, if it is present -> the JSON file must be used for credentials.
+	if dir, isSet := os.LookupEnv(prefixString + absCredentialDirectory); isSet {
+		jsonCredentialFile, err := findFileWithExtensionInDir(dir, ".json")
+		if err != nil {
+			return "", "", fmt.Errorf("error while finding a JSON credential file in %v directory with error: %w", dir, err)
+		}
+		if jsonCredentialFile != "" {
+			credentials, err := readABSCredentialsJSON(jsonCredentialFile)
+			if err != nil {
+				return "", "", fmt.Errorf("error getting credentials using %v JSON file in a directory with error: %w", jsonCredentialFile, err)
+			}
+			return credentials.StorageAccount, credentials.SecretKey, nil
+		}
+		// Non JSON credential files might exist in the credential directory, do not return
+	}
+
```

Comment:
I would suggest put this block of code before checking the `os.LookupEnv(prefixString + absCredentialJSONFile)`. This will save you from extra part of code.
```
See if `absCredentialDirectory` is set, if yes -> Check if `jsonCredentialFile` is present, if yes --> set the ENV `absCredentialJSONFile` and continue the code as it is
```
By this you don't have to handle to write piece of code:
```
		if jsonCredentialFile != "" {
			credentials, err := readABSCredentialsJSON(jsonCredentialFile)
			if err != nil {
				return "", "", fmt.Errorf("error getting credentials using %v JSON file in a directory with error: %w", jsonCredentialFile, err)
			}
			return credentials.StorageAccount, credentials.SecretKey, nil
		}
```

Reply:
I'd specifically not chosen to do this because the environment variable is set by the consumer of etcd-backup-restore.

etcd-backup-restore changing the environment variables that it consumes from the user as a part of configuration does not feel right to me.

Let's say that a consumer of etcd-druid provides a JSON file as a part of the credential secret. During a credential rotation, they choose to provide the credentials as normal data fields which are mounted as individual files in the etcd-backup-restore container. If etcd-backup-restore sets the JSON environment variable, it will not be unset when this credential rotation happens, and the credentials will still be looked for as a JSON file.

Due to configuration to the binary being changed by the binary causing possible confusion, and the above case, I wrote the code as it is right now.

If you still feel the flow would be made simpler due to the suggestion, I will change the code.

Reply:
good edge catch but can be easily handled by this:
```
See if `absCredentialDirectory` is set, if yes -> Check if `jsonCredentialFile` is present, if yes --> set the ENV `absCredentialJSONFile` and continue the code as it is else unset ENV for JSON.
```
And secret rotation detection shouldn't depend on any `ENV`, it should list all files and check if any file get updated or not.

Reply:
But I leave this decision to you, basically my point was to make less code change.

Reply:
Alright, I'll take a call on this. I don't mind the larger change since it's easier to follow.